python/paddle/v2/fluid/executor.py

```diff
@@ -146,33 +146,76 @@ def run(self,
 
         program = program.clone()
         global_block = program.global_block()
-        feed_var = global_block.create_var(
-            name=feed_var_name,
-            type=core.VarDesc.VarType.FEED_MINIBATCH,
-            persistable=True)
-
-        for i, name in enumerate(feed):
-            out = global_block.var(name)
-            global_block.prepend_op(
-                'feed',
-                inputs={'X': [feed_var]},
-                outputs={'Out': [out]},
-                attrs={'col': i})
-            cur_feed = feed[name]
-            if not isinstance(cur_feed, core.LoDTensor):
-                cur_feed = self.aslodtensor(cur_feed)
-            core.set_feed_variable(scope, cur_feed, feed_var.name, i)
-
-        fetch_var = global_block.create_var(
-            name=fetch_var_name,
-            type=core.VarDesc.VarType.FETCH_LIST,
-            persistable=True)
-        for i, var in enumerate(fetch_list):
-            global_block.append_op(
-                type='fetch',
-                inputs={'X': [var]},
-                outputs={'Out': [fetch_var]},
-                attrs={'col': i})
+
+        if feed_var_name in global_block.vars:
+            feed_var = global_block.var(feed_var_name)
+        else:
+            feed_var = global_block.create_var(
+                name=feed_var_name,
+                type=core.VarDesc.VarType.FEED_MINIBATCH,
+                persistable=True)
+
+        if fetch_var_name in global_block.vars:
+            fetch_var = global_block.var(fetch_var_name)
+        else:
+            fetch_var = global_block.create_var(
+                name=fetch_var_name,
+                type=core.VarDesc.VarType.FETCH_LIST,
+                persistable=True)
+
+        feed_count = 0
+        fetch_count = 0
+        for op in global_block.ops:
+            if op.desc.type() == 'feed':
+                feed_count += 1
+                assert op.desc.input('X')[0] == feed_var_name
```

Comment:
Do we need to keep all the input variable of `feed_op` the same name? I'm not sure...

Reply:
I assume there will be only one `feed_variable` in the program Desc, which holds a vector of Tensor. 
Each feed operator will take one of the Tensor data from this input `feed_variable` and copy the data to its output variable. So I think different feed operator should have different output variable names (`x`, `y`, etc), but should have the same input variable which is our `feed_var_name` here.  

Could you explain a little more about your point? Thanks!

Reply:
Well, I am not sure, whether the input data can be feed correctly if we set the `feed_holder_name` to different string... and do we need to support this...

Reply:
I prefer to make the feed_holder_name (same for fetch) consistent so that the code is simpler and less error prone. 

Even if this feature is later found out to be desirable, we can fix it in future PR. What do you think?